{
 "metadata": {
  "name": "",
  "signature": "sha256:af8b3faad86d7d61bea5e165b9db52b65c305c1a32691216b28abf2194f483a7"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Using IGS to do alpha-diversity"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "This notebook includes many experiments I have tried. Many of them did not lead to meaningful results. But, this is how research works. :) \n",
      "\n",
      "For the whole picture about doing diversity analysis to metagenomics samples using IGS, please check: <http://nbviewer.ipython.org/github/qingpeng/2013-diversity/blob/master/notebook/2013-diversity-summary.ipynb>"
     ]
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "pipeline to do alpha diversity analysis of simulated ecoli reads dataset"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "### 1. get simulated reads\n",
      "\n",
      "- 50X and 150X\n",
      "- 1%, 2% and 0% error rate\n",
      "\n",
      "git clone https://github.com/ctb/dbg-graph-null.git\n",
      "\n",
      "e.coli_reference.fa from 2013-khmer-counting repo\n",
      "\n",
      "    python dbg-graph-null/make-reads.py -C 150 ~/Dropbox/Manuscript/2013-diversity/pipeline/e.coli_reference.fa  >e.coli_150x_0.01e_100.fa\n",
      "    python dbg-graph-null/make-reads.py -C 50 ~/Dropbox/Manuscript/2013-diversity/pipeline/e.coli_reference.fa  >e.coli_50x_0.01e_100.fa\n",
      "    python dbg-graph-null/make-reads.py -e 0.00 -C 50 ~/Dropbox/Manuscript/2013-diversity/pipeline/e.coli_reference.fa  >e.coli_50x_0.00e_100.fa\n",
      "    python dbg-graph-null/make-reads.py -e 0.02 -C 50 ~/Dropbox/Manuscript/2013-diversity/pipeline/e.coli_reference.fa  >e.coli_50x_0.02e_100.fa\n",
      "\n",
      "### 2. create graph file\n",
      "\n",
      "print_load_sh.sh:\n",
      "\n",
      "    for i in *.fa; do echo load-into-counting.py -x 1e8 -k 20 $i.kh $i \\&;  done\n",
      "\n",
      "    bash print_load_sh.sh  >load-into-counting.sh\n",
      "    bash load-into-counting.sh\n",
      "\n",
      "### 3. generate config.txt for counting reads coverage spectrum\n",
      "\n",
      "    [qingpeng@dev-intel14 Ecoli_Alpha]$ more config.txt \n",
      "    e.coli_150x_0.01e_100.fa.kh  e.coli_50x_0.00e_100.fa.kh  e.coli_50x_0.01e_100.fa.kh  e.coli_50x_0.02e_100.fa.kh\n",
      "    e.coli_150x_0.01e_100.fa  e.coli_50x_0.00e_100.fa  e.coli_50x_0.01e_100.fa  e.coli_50x_0.02e_100.fa\n",
      "    10000000000\n",
      "\n",
      "### 4. count reads coverage spectrum across species for each reads file(get .comb)\n",
      "\n",
      "    python ~/Dropbox/Manuscript/2013-diversity/scripts/get_comb_multi.py config.txt &\n",
      "\n",
      "### 5. get .comb list file and get spectrum frequency across samples\n",
      "\n",
      "    ls *.comb >comb.list\n",
      "    python ~/Dropbox/Manuscript/2013-diversity/scripts/count_spectrum_freq_multiple_files.py comb.list e.coli.spectrum   \n",
      "\n",
      "### 6. generate MAP file (prepare for QIIME run) (also used in seperate_IGS.py)\n",
      "\n",
      "e.coli_Map.txt:\n",
      "\n",
      "    #SampleID\tBarcodeSequence\tLinkerPrimerSequence\tDescription coverage  error_rate  file_name\n",
      "    sampleA\tA\tA\te.coli_150x_0.01e_100 150x  0.01  e.coli_150x_0.01e_100.fa.comb\n",
      "    sampleB\tA\tA\te.coli_50x_0.00e_100  50x 0.00  e.coli_50x_0.00e_100.fa.comb\n",
      "    sampleC\tA\tA\te.coli_50x_0.01e_100  50x 0.01  e.coli_50x_0.01e_100.fa.comb\n",
      "    sampleD A A e.coli_50x_0.02e_100  50x 0.02  e.coli_50x_0.02e_100.fa.comb\n",
      "\n",
      "### 7. get .IGS and .IGS_abund files from *.spectrum file, listing all the IGSs and get the IGS-abundance matrix\n",
      "\n",
      "    python ~/Dropbox/Manuscript/2013-diversity/scripts/seperate_IGS.py e.coli.spectrum e.coli_Map.txt \n",
      "\n",
      "\n",
      "### 8. convert .IGS file to .biom format\n",
      "\n",
      "    module load QIIME/1.7.0\n",
      "    biom convert -i e.coli.spectrum.IGS -o e.coli.spectrum.IGS.biom --table-type=\"OTU table\"\n",
      "\n",
      "### 9. get summary from .biom file\n",
      "\n",
      "    biom summarize-table -i  e.coli.spectrum.IGS.biom -o e.coli.spectrum.IGS.biom.summary.txt\n",
      "\n",
      "    [qingpeng@dev-intel14 Ecoli_Alpha]$ more e.coli.spectrum.IGS.biom.summary.txt\n",
      "    Num samples: 4\n",
      "    Num observations: 260333\n",
      "    Total count: 2105023\n",
      "    Table density (fraction of non-zero values): 0.408\n",
      "    Table md5 (unzipped): 6b71f7e4f1d3957fb47f669a0231de6f\n",
      "\n",
      "    Counts/sample summary:\n",
      "     Min: 310594.0\n",
      "     Max: 1016129.0\n",
      "     Median: 389150.000\n",
      "     Mean: 526255.750\n",
      "     Std. dev.: 287829.236\n",
      "     Sample Metadata Categories: None provided\n",
      "     Observation Metadata Categories: None provided\n",
      "\n",
      "    Counts/sample detail:\n",
      "     sampleB: 310594.0\n",
      "     sampleC: 328731.0\n",
      "     sampleD: 449569.0\n",
      "     sampleA: 1016129.0\n",
      " \n",
      " \n",
      "### 10. decide min max step for saturation curve, run .qsub with QIIME scripts\n",
      "\n",
      "    qsub ~/Dropbox/Manuscript/2013-diversity/pipeline/e_coli_alpha_diversity.qsub\n",
      " \n",
      " \n",
      " "
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      " new method (treat each IGSs seperately totally)\n",
      " ------\n",
      " \n",
      " From step 7:\n",
      " \n",
      " treat IGSs in different seperately\n",
      " \n",
      " 7. \n",
      "python ~/Dropbox/Manuscript/2013-diversity/scripts/seperate_IGS_for_alpha.py e.coli.spectrum e.coli_Map.txt\n",
      "\n",
      "8. \n",
      "biom convert -i e.coli.spectrum.IGS.alpha -o e.coli.spectrum.IGS.alpha.biom --table-type=\"OTU table\"\n",
      "\n",
      "9.\n",
      "biom summarize-table -i e.coli.spectrum.IGS.alpha.biom -o e.coli.spectrum.IGS.alpha.biom.summary.txt\n",
      "10.\n",
      "\n",
      "qsub ~/Dropbox/Manuscript/2013-diversity/pipeline/e_coli_alpha_diversity_for_alpha_biom.qsub\n",
      "\n",
      " \n",
      "file:///Users/qingpeng/Google%20Drive/Dropbox/Manuscript/2013-diversity/Data/rare_plot_alpha/rarefaction_plots.html\n",
      "\n",
      "\n",
      "\n",
      "beginning of curve:\n",
      "\n",
      "/mnt/home/qingpeng/Manuscript/2013-diversity/data/Ecoli_Alpha/Begin_curve"
     ]
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "new method (removing low abundance k-emrs with filter_abund.py firstly)"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "filter-abund.py -C 2 e.coli_50x_0.00e_100.fa.kh e.coli_50x_0.00e_100.fa -V\n",
      "filter-abund.py -C 2 e.coli_50x_0.01e_100.fa.kh e.coli_50x_0.01e_100.fa -V \n",
      "filter-abund.py -C 2 e.coli_50x_0.02e_100.fa.kh e.coli_50x_0.02e_100.fa -V\n",
      "\n",
      "    DONE writing.\n",
      "    processed 2319837 / wrote 2041907 / removed 277930\n",
      "    processed 231983700 bp / wrote 167487805 bp / removed 64495895 bp\n",
      "    discarded 27.8%\n",
      "    output in e.coli_50x_0.01e_100.fa.abundfilt\n",
      "    done loading in sequences\n",
      "    DONE writing.\n",
      "    processed 2319837 / wrote 1915430 / removed 404407\n",
      "    processed 231983700 bp / wrote 144771412 bp / removed 87212288 bp\n",
      "    discarded 37.6%\n",
      "    output in e.coli_50x_0.02e_100.fa.abundfilt\n"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "    for i in *.fa.abundfilt; do echo load-into-counting.py -x 1e8 -k 20 $i.kh $i \\&;  done >load-into-counting-abund.py\n",
      "    bash load-into-counting-abund.py\n",
      "    \n",
      "config-abund.txt:\n",
      "\n",
      "    e.coli_50x_0.01e_100.fa.abundfilt.kh  e.coli_50x_0.02e_100.fa.abundfilt.kh\n",
      "    e.coli_50x_0.01e_100.abundfilt.fa  e.coli_50x_0.02e_100.abundfilt.fa\n",
      "    10000000000\n",
      "\n"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "    [qingpeng@dev-intel14 Ecoli_Alpha]$ cat >get_comb_multi.qsub\n",
      "    #!/bin/bash -login\n",
      "    #PBS -l walltime=2:00:00,nodes=01:ppn=1,mem=5gb\n",
      "    #PBS -q main\n",
      "    #PBS -M qingpeng@gmail.com\n",
      "    #PBS -m abe\n",
      "    #PBS -A ged-intel11\n",
      "\n",
      "    module load khmer\n",
      "    module load screed\n",
      "    cd $PBS_O_WORKDIR\n",
      "    python ~/Dropbox/Manuscript/2013-diversity/scripts/get_comb_multi.py config-abund.txt\n",
      "    ^C\n",
      "    [qingpeng@dev-intel14 Ecoli_Alpha]$ qsub get_comb_multi.qsub\n",
      "\n",
      "\n",
      "\n",
      "    [qingpeng@dev-intel10 Ecoli_Alpha]$ cat >comb-abund.list\n",
      "    e.coli_50x_0.01e_100.fa.abundfilt.comb\n",
      "    e.coli_50x_0.02e_100.fa.abundfilt.comb\n",
      "\n",
      "\n",
      "    python ~/Dropbox/Manuscript/2013-diversity/scripts/count_spectrum_freq_multiple_files.py comb-abund.list e.coli-abund.spectrum\n",
      "\n",
      "    [qingpeng@dev-intel10 Ecoli_Alpha]$ cat >e.coli_Map-abund.txt\n",
      "    #SampleID\tBarcodeSequence\tLinkerPrimerSequence\tDescription\tcoverage\terror_rate\tfile_name\n",
      "    sampleE\tA\tA\te.coli_50x_0.01e_100-abund\t50x\t0.01\te.coli_50x_0.01e_100.fa.abundfilt.comb\n",
      "    sampleF\tA\tA\te.coli_50x_0.02e_100-abund\t50x\t0.02\te.coli_50x_0.02e_100.fa.abundfilt.comb\n",
      "\n",
      "    python ~/Dropbox/Manuscript/2013-diversity/scripts/seperate_IGS_for_alpha.py e.coli-abund.spectrum e.coli_Map-abund.txt\n",
      "    biom convert -i e.coli-abund.spectrum.IGS.alpha -o e.coli-abund.spectrum.IGS.alpha.biom --table-type=\"OTU table\"\n",
      "    biom summarize-table -i e.coli-abund.spectrum.IGS.alpha.biom -o e.coli-abund.spectrum.IGS.alpha.biom.summary.txt\n",
      " \n",
      "Prepare for plotting:\n",
      "\n",
      "    cd Filter_Abund/\n",
      "    ln -fs ../e.coli_Map-abund.txt .\n",
      "    ln -fs ../e.coli-abund.spectrum.IGS.alpha.biom .\n",
      "    qsub plot_curve.qsub\n",
      " \n",
      " \n",
      "plotting in directory: rare_plot_alpha_filter_abund\n",
      "\n",
      " "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}